library/alloc/src/rc.rs

```diff
@@ -403,7 +419,14 @@ impl<T> Rc<T> {
     /// impl Gadget {
     ///     /// Construct a reference counted Gadget.
     ///     fn new() -> Rc<Self> {
-    ///         Rc::new_cyclic(|me| Gadget { me: me.clone() })
+    ///         // `me` is an uninitialized `Weak` at the start of the closure.
+    ///         Rc::new_cyclic(|me| {
+    ///             // Create the actual struct here.
+    ///             Gadget { me: me.clone() }
+    ///
+    ///             // Before returning, `new_cyclic` increase the "strong"
+    ///             // counter of the `Rc` by one.
```

Comment:
I'm not sure if we should explain this implementation detail in this example. I'm afraid this might just confuse the reader.